### PR TITLE
[CPU] Fix alloc-size-larger-than warning.

### PR DESCRIPTION
**Pull request by nshchego**
### Details:
 - [CPU] Suppress alloc-size-larger-than warning

### Tickets:
 - CVS-140251

### CODE REVIEW

File: src/plugins/intel_cpu/CMakeLists.txt [hunk 1]
```diff
@@ -9,7 +9,7 @@ endif()
 set(TARGET_NAME "openvino_intel_cpu_plugin")
 
 if((CMAKE_COMPILER_IS_GNUCXX OR OV_COMPILER_IS_CLANG) AND CMAKE_CXX_STANDARD GREATER_EQUAL 20)
-    set(CMAKE_CXX_FLAGS "-Wno-error=deprecated ${CMAKE_CXX_FLAGS}")
+    set(CMAKE_CXX_FLAGS "-Wno-error=deprecated -Wno-alloc-size-larger-than ${CMAKE_CXX_FLAGS}")
```
Comment (ilya-lavrenov) [inline src/plugins/intel_cpu/CMakeLists.txt:12]: I suppose it's available only with newer compiler version? should be suppress it conditionally? maybe we can suppress inside the code directly to localize suppression?

Comment (nshchego) [reply src/plugins/intel_cpu/CMakeLists.txt]: Fixed in code instead

**Comment by ilya-lavrenov** [comment]
@EgorDuplensky @maxnick 
Could you please review? We need to merge rather soon than later to unblock Ubuntu 24.04 support.

File: src/plugins/intel_cpu/src/cpu_memory.cpp [hunk 2]
```diff
@@ -398,7 +398,11 @@ StringMemory::OvString* StringMemory::StringMemoryMngr::getStringPtr() const noe
 bool StringMemory::StringMemoryMngr::resize(size_t size) {
     bool sizeChanged = false;
     if (size > m_str_upper_bound) {
-        auto ptr = new OvString[size];
+        if (size > PTRDIFF_MAX) {
+            OPENVINO_THROW("Requested allocation size { ", size, " } exceeds PTRDIFF_MAX.");
+        }
+        int64_t new_size = static_cast<int64_t>(size); // WA for warning alloc-size-larger-than
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/cpu_memory.cpp:404]: please use `ptrdiff_t`

Comment (nshchego) [reply src/plugins/intel_cpu/src/cpu_memory.cpp]: done

**Review by maxnick — APPROVED** [review]

**Pull request merged**